src/services/common.ts

```diff
@@ -2,9 +2,9 @@ export function isString(obj: unknown): obj is string {
   return typeof obj === 'string';
 }
 
-export const safeParse = (json: string) => {
+export const safeParse = (json: string | undefined) => {
```

Comment:
Nit. This should be equivalent:

```suggestion
export const safeParse = (json?: string) => {
```